### ISSUE TITLE
Any option to include python binary in pex bundle

### ISSUE THREAD

**Issue by vnkesarwani** [criteo/cluster-pack#64]
Hi,

I have a use-case where python interpreter will not be available on all the nodes. Is it possible to bundle python binary along with pex and execute in environment where python interpreter is not present?? I know for those cases we may need to use freezers..

Any suggestion

Thanks,

**Comment by fhoering** [comment]
This is currently not possible with pex. It would be a nice feature to add to [pex](https://github.com/pantsbuild/pex) directly to get something similar to what conda provides. 

But it is not so easy to implement as the core point of zipped files is to set the python interpreter as a shebang. So the python interpreter needs to be available form the beginning. 
It can maybe work with having one single python installed on the system which then forwards to another interpreter but this seems ugly. Maybe it is possible to provide as bash script as shebang that then forwards to the right embedded python interpreter but this needs a PoC first.

**Comment by vnkesarwani** [comment]
Yeah, you are right. It will be awesome if we can bundle python binary.

Do you have any suggestion to change pex script to bundle python binary along with pex??
